winpr/libwinpr/clipboard/dirent.c

```diff
@@ -0,0 +1,91 @@
+
+#include <stdio.h>
+#include "dirent.h"
+
+#include "../log.h"
+#define TAG WINPR_TAG("clipboard.synthetic.file")
+
+DIR* opendir(const char* name)
+{
+	DIR* dir = NULL;
+	WIN32_FIND_DATAW FindData;
+	char namebuf[MAX_PATH];
+	WCHAR* file = NULL;
+	int nLen = 0;
+	int nRet = 0;
+
+	sprintf(namebuf, "%s\\*.*", name);
+	nLen = strlen(namebuf);
+
+	do
+	{
+		nRet = ConvertToUnicode(CP_UTF8, 0, namebuf, -1, file, 0);
+		if (nRet != nLen)
+			break;
+
+		HANDLE hFind = FindFirstFileW(file, &FindData);
+		if (hFind == INVALID_HANDLE_VALUE)
+		{
+			WLog_ERR(TAG, "FindFirstFile failed (%d)\n", GetLastError());
+			break;
+		}
+
+		dir = (DIR*)calloc(1, sizeof(DIR));
+		if (!dir)
+		{
+			WLog_ERR(TAG, "DIR memory allocate fail\n");
+			break;
+		}
+
+		dir->dd_fd = hFind; // simulate return
+	} while (0);
+
+	if (file)
+		free(file);
+
+	return dir;
+}
+
+struct dirent* readdir(DIR* d)
+{
+	int i;
+
+	BOOL bf;
+	WIN32_FIND_DATAW FileData;
+	if (!d)
+		return 0;
+
+	bf = FindNextFileW(d->dd_fd, &FileData);
+	// fail or end
+	if (!bf)
+		return 0;
+
+	struct dirent* dirent =
+	    (struct dirent*)calloc(1, sizeof(struct dirent) + sizeof(FileData.cFileName));
+
+	for (i = 0; i < 256; i++)
+	{
+		dirent->d_name[i] = FileData.cFileName[i];
```

Comment:
This should be converted from 16bit unicode to `utf-8` or am I missing something?